### PR TITLE
operator: change default MachineInventory name

### PR DESCRIPTION
**Pull request by fgiudici**
Right now when the MachineRegistration is missing a 'spec.MachineName' value we set the following hostname:
m-${System Information/Manufacturer}-${System Information/Product Name}-${System Information/UUID}
    
This causes an invalid hostname if config.registration.no-smbios is set to true, failing the registration.
When no-smbios is set to false instead, we may still fail the registration as the hostname derived by no-smbios data is truncated to 58 chars, which may end up truncating the UUID, which is the only parameter ensuring uniqueness.
    
So, when SMBIOS data is available (no-smbios == false) the default MachineInventory name got the UUID parameter as the first one (so it will not be truncated); when no SMBIOS is are available instead, the hostname will be set to a random DUID.
    
Fixes: https://github.com/rancher/elemental-operator/issues/144

### CODE REVIEW

**Review by kkaempf — CHANGES_REQUESTED** [review]
Looks ok to me.
Please add a test case !

File: pkg/server/register.go [hunk 1]
```diff
@@ -36,7 +37,7 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-const defaultName = "m-${System Information/Manufacturer}-${System Information/Product Name}-${System Information/UUID}"
```
[the remark this answers is not in the document]
Comment (fgiudici) [reply pkg/server/register.go:39]: I prefer too to have manufacturer and product first in the name. The issue is that in order to have a unique name we have to ensure that the UUID is not truncated (yep, Product Name can become long and already happened a clash in the name as the UUID got truncated).
We can truncate Manufacturer + Product Name as the UUID lenght is known... but that would need to rework the build name function. I'm lazy, you got me 🤣 - will rework it.

**Comment by Itxaka** [comment]
we probably should just use the UUID as name? Not sure what the product name and manufacturer provide here, we could simply auto add some of those values as labels directly on the registration using the smbios data like `elemental.cattle.io/manufacturer: X` instead.

In any case, just having the uuid which should be unique as a default name if the registration does not provide an override, looks good to me.

File: pkg/server/register.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fgiudici) [reply pkg/server/register.go:41]: well, in the end resorted to keep the UUID only (m-$UUID format): as this is the default, means who registers it doesn't care about the name. Let's ensure this is unique and keep it simple (and also coherent for machines without smbios data).

File: pkg/server/register.go [hunk 2]
```diff
@@ -123,12 +124,22 @@ func (i *InventoryServer) unauthenticatedResponse(machineRegistration *elm.Machi
 }
 
 func (i *InventoryServer) createMachineInventory(req *http.Request, inventory *elm.MachineInventory, registration *elm.MachineRegistration) (*elm.MachineInventory, error) {
+	noSMBIOS := registration.Spec.Config.Elemental.Registration.NoSMBIOS
+
 	inventory.Name = registration.Spec.MachineName
 	if inventory.Name == "" {
-		inventory.Name = defaultName
+		if noSMBIOS {
+			inventory.Name = "m-" + uuid.NewString()
```
[the remark this answers is not in the document]
Comment (fgiudici) [reply pkg/server/register.go]: now we keep the format coherent for both no-smbios==true and false:
`m-{UUID}`

**Comment by mjura** [comment]
Please rebase

**Review by mjura — APPROVED** [review]
LGTM

**Review by Itxaka — APPROVED** [review]

**Comment by fgiudici** [comment]
> Looks ok to me. Please add a test case !
Added unit tests

**Comment by Itxaka** [comment]
@fgiudici remeber the docs! they need to be changed as the default has changed!

**Comment by fgiudici** [comment]
> @fgiudici remeber the docs! they need to be changed as the default has changed!

🤦🏼 thanks, I forgot! Will fix now 😄

**Pull request merged**